_posts/2020-10-24-text-layout.md

```diff
@@ -0,0 +1,217 @@
+---
+layout: post
+title:  "Text layout is a loose hierarchy of segmentation"
+date:   2020-10-25 15:00:42 -0700
+categories: [text]
+---
+I love text layout, and have been working with it in one form or other for over 35 years. Yet, knowledge about it is quite arcane. I don't believe there is a single place where it's all properly written down. I have some explanation for that: while basic text layout is very important for UI, games, and other contexts, a lot of the "professional" needs around text layout are embedded in *much* more complicated systems such as Microsoft Word or a modern Web browser.
+
+A complete account of text layout would be at least a small book. Since there's no way I can write that now, this blog post is a small step towards that - in particular, an attempt to describe the "big picture," using the conceptual framework of a "loose hierarchy." Essentially, a text layout engine breaks the input into finer and finer grains, then reassembles the results into a text layout object suitable for drawing, measurement, and hit testing.
+
+The main hierarchy is concerned with laying out a single strip of text. Line breaking is also important, but has a separate, parallel hierarchy.
```

Comment:
I think it might be worth defining 'strip', or at least clarifying why we don't say 'line'.